### ISSUE TITLE
Aggregated by Experiment: Change file names

### ISSUE THREAD

**Issue by dvenprasad** [AlexsLemonade/refinebio#610]
### Context

This came up when @jaclyn-taroni and I were chatting about illustrations for download files (for docs)

### Problem or idea

Currently, all the files are just dumped in the download folder. If someone downloads 10 experiments, it will be messy for them find their files. The files should be grouped together in folders based on Experiment. 

For example:
![agg-experiments](https://user-images.githubusercontent.com/15315514/45447294-18620d80-b69d-11e8-9f07-496236ba8361.png)

Related to #609. But this should be given a higher priority. 

### Solution or next step

Tagging @Miserlou for input on how best move this forward

**Comment by jaclyn-taroni** [comment]
Grabbed data from the dev stack this morning. The structure looks good. File names are a bit different. 
```
single_species_by_experiment_example/
├── E-GEOD-39842
│   ├── E-GEOD-39842_metadata.tsv
│   └── E-GEOD-39842.tsv
├── LICENSE.TXT
├── metadata.json
├── README.md
├── SRP062067
│   └── SRP062067_metadata.tsv
└── SRP111705
    └── SRP111705_metadata.tsv

3 directories, 7 files
```
`metadata.json` -> `aggregated_metadata.json`
`SRP111705_metadata.tsv` -> `metadata_SRP111705.tsv`

**Comment by cgreene** [comment]
Is there a reason to put the word metadata first? I am imagining a world in which I am sorting by file name, and it seems like having the experiment/metadata pair show up together (instead of all the experiments followed by all the metadata) would be preferred. I may not be thinking of why having `metadata_` first is better though. I realize that is the way it is designed, but I'm just trying to think about why.

**Comment by jaclyn-taroni** [comment]
Question for @dvenprasad

**Comment by Miserlou** [comment]
That's what I was thinking @casey

**Comment by jaclyn-taroni** [comment]
I will say after working with a bunch of this data this weekend with the new folder structure, I do think putting `metadata` first in the file name would reduce some friction.

**Comment by dongbohu** [comment]
@jaclyn-taroni and @dvenprasad: So why when the dataset is aggregated by species, we want to have species-level json file, but when it is aggregated by experiment, we don't need experiement-level json file?

**Comment by dvenprasad** [comment]
The folder structure already groups the files by experiment. Adding ```meatadata``` first will help them distinguish between the files for the same experiment easily.

**Comment by dongbohu** [comment]
Another question for @jaclyn-taroni and @dvenprasad: since we are going to rename all metadata files in `metadata_*` format, when the aggregation is not set in a dataset, should we name the file as:
`ALL/metadata_ALL.tsv` instead of `ALL/ALL_metadata.tsv`?

**Comment by jaclyn-taroni** [comment]
@dongbohu I'm not sure what you mean here
> when the aggregation is not set in a dataset

**Comment by dongbohu** [comment]
@jaclyn-taroni: Let me give you a little more context. According to:
https://github.com/AlexsLemonade/refinebio/blob/b4a768b0c5ac53ba76144f8421cdf0d97c750c54/common/data_refinery_common/models/models.py#L853
`aggregate_by` in `Dataset` model has three choices (also see lines 831-835 of the same file):
- ALL
- EXPERIMENT
- SPECIES

This issue is about `EXPERIMENT`, and #609 is about `SPECIES`. When `aggregate_by` is `ALL`, the smasher will simply save all samples metadata in one file. See the logic here:
https://github.com/AlexsLemonade/refinebio/blob/b4a768b0c5ac53ba76144f8421cdf0d97c750c54/workers/data_refinery_workers/processors/smasher.py#L274

My question is, since we decide to name all metadata tsv files as `metadata_*.tsv`, should we also name the tsv file when `aggregate_by` is `ALL` into `metadata_ALL.tsv`? In another word, should we change this line:
 https://github.com/AlexsLemonade/refinebio/blob/b4a768b0c5ac53ba76144f8421cdf0d97c750c54/workers/data_refinery_workers/processors/smasher.py#L277
into:
```with open(all_dir + 'metadata_ALL.tsv', 'w') as output_file:```

What do you think of my other question?
https://github.com/AlexsLemonade/refinebio/issues/609#issuecomment-424062084

Please let me know if you need further clarification.

**Comment by dongbohu** [comment]
Implemented in #648.
